crates/subnetworks/beacon/src/network.rs

```diff
@@ -77,39 +78,123 @@ impl BeaconNetwork {
-            Some(trusted_block_root) => Some(trusted_block_root),
-            None => {
-                // If no trusted block root is provided, we check for the latest block root in the
-                // database.
-                let block_root = storage_clone.read().lookup_latest_block_root()?;
-                if let Some(block_root) = block_root {
-                    info!(block_root = %block_root, "No trusted block root provided. Using latest block root from storage.");
+        tokio::spawn(async move {
+            let beacon_sync = BeaconSync::new(overlay_tx);
+            let beacon_sync = beacon_sync.start(trusted_block_root).await;
+            match beacon_sync {
+                Ok(client) => {
+                    let mut beacon_client = beacon_client_clone.lock().await;
+                    *beacon_client = Some(client);
                 }
-                block_root
-            }
-        };
-
-        if let Some(trusted_block_root) = trusted_block_root {
-            tokio::spawn(async move {
-                let beacon_sync = BeaconSync::new(overlay_tx);
-                let beacon_sync = beacon_sync.start(trusted_block_root).await;
-                match beacon_sync {
-                    Ok(client) => {
-                        let mut beacon_client = beacon_client_clone.lock().await;
-                        *beacon_client = Some(client);
-                    }
-                    Err(err) => {
-                        error!(error = %err, "Failed to start beacon sync.");
-                    }
+                Err(err) => {
+                    error!(error = %err, "Failed to start beacon sync.");
                 }
-            });
-        }
+            }
+        });
 
         Ok(Self {
             overlay: Arc::new(overlay),
             beacon_client,
         })
     }
 }
+
+/// Get the trusted block root to start syncing light client from.
+fn get_trusted_block_root(
+    portal_config: &PortalnetConfig,
+    storage: Arc<PLRwLock<BeaconStorage>>,
+) -> anyhow::Result<B256> {
+    // 1) Check if a trusted block root was provided via config
+    if let Some(block_root) = portal_config.trusted_block_root {
+        return Ok(block_root);
+    }
+
+    // 2) Otherwise, try to read the latest block root from storage
+    let maybe_db_block_root = storage.read().lookup_latest_block_root()?;
+    if let Some(db_block_root) = maybe_db_block_root {
+        info!(
+            block_root = %db_block_root,
+            "No trusted block root provided. Using latest block root from storage."
+        );
+        return Ok(db_block_root);
+    }
+
+    // 3) If there's still nothing, log and attempt to load the default checkpoint file
+    info!("No trusted block root provided and no block root found in storage. Loading default checkpoint.");
+    let block_root = parse_trusted_block_root(DEFAULT_TRUSTED_BLOCK_ROOT.trim())?;
+
+    Ok(block_root)
```

Comment:
```suggestion
    info!("No trusted block root provided and no block root found in storage. Loading default checkpoint.");
    parse_trusted_block_root(DEFAULT_TRUSTED_BLOCK_ROOT.trim())
```

Or if you apply other suggestion, than simply:

```suggestion
    info!("No trusted block root provided and no block root found in storage. Loading default checkpoint.");
    get_default_trusted_root()
```